models/crypto/products.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * This file contains wrapper functions to interact with the DB.
+ * This will contain the DB schema if we start consuming an ORM for managing the DB operations
+ */
+
+const logger = require('../../utils/logger')
+const firestore = require('../../utils/firestore')
+const productsModel = firestore.collection('crypto/shop/products')
+
+/**
+ * Fetches the data of products in crypto site
+ * @return {Promise<productsModel|Array>}
+ */
+
+const fetchProducts = async () => {
+  try {
+    const snapshot = await productsModel.get()
+
+    const allProducts = []
+
+    snapshot.forEach((doc) => {
+      allProducts.push({
+        id: doc.id,
+        ...doc.data()
+      })
+    }
+    )
+
+    return allProducts
+  } catch (err) {
+    logger.error('Error retrieving products data', err)
+    throw err
+  }
+}
+
+/**
+ * Fetches the data of a product in crypto site
+ * @return {Promise<productsModel|Object>}
+ */
+
+const fetchProduct = async (id) => {
+  try {
+    const product = await productsModel.doc(id).get()
+    if (product.exists) {
+      return {
+        product: {
+          id: id,
+          ...product.data()
+        }
+      }
+    }
+    return ''
+  } catch (err) {
+    logger.error('Error retrieving product data', err)
+    throw err
+  }
+}
+
+/**
+ * Adds the data of a product to crypto site
+ * @return {Promise<productsModel|Object>}
+ */
+
+const addProduct = async (productData) => {
+  try {
+    let productInfo = await productsModel.where('productId', '==', productData.productId).limit(1).get()
+
+    if (productInfo.empty) {
+      productInfo = await productsModel.doc().set(productData)
+      return { productId: productInfo.id }
+    }
+
+    return ''
```

Comment:
This sort of return statement doesn't seem right, return some error message or come up with some better approach?

Reply:
This would return the product if found or else it would return an empty string. We can remove the return statement which means it would return undefined which is the same, but I think this is easy to understand, If not then let me know I will change it 😀

Reply:
Instead of empty string, can we have message that product was not found.

Reply:
That has been taken care of in the controller.js file [`here`](https://github.com/Real-Dev-Squad/website-backend/pull/72/files/184c32c5c2771fd61aa937ac213474fbaafbc21f#diff-1bf082f4fa1b819960342e413ebff7efc8893704defa73f4cf3ed3e434cd1a6eR60-R72)